src/cookbook/packaging-components-npm.md

```diff
@@ -0,0 +1,197 @@
+  },
+  "devDependencies": {
+    "rollup": "^1.17.0",
+    "@rollup/plugin-buble": "^0.21.3",
+    "@rollup/plugin-commonjs": "^11.1.0",
+    "rollup-plugin-vue": "^5.0.1",
+    "vue": "^2.6.10",
+    "vue-template-compiler": "^2.6.10"
+    ...
+  },
+  ...
+}
+```
+
+Our changes to `package.json` are complete. Next, we need a small wrapper to export the actual SFC, plus a minimal Rollup configuration, and we're set!
+
+### What does my packaged component look like?
+
+Depending on how your component is being used, it needs to be exposed as either a [CommonJS/UMD](https://medium.freecodecamp.org/javascript-modules-a-beginner-s-guide-783f7d7a5fcc#c33a) javascript module, an [ES6 javascript](https://medium.freecodecamp.org/javascript-modules-a-beginner-s-guide-783f7d7a5fcc#4f5e) module so it's immediately available to the page. This is accomplished by a `wrapper.js` file which handles the module export and auto-install. That wrapper, in its entirety, looks like this:
+
+```js
+// Import vue component
+import component from './my-component.vue'
+
+// Declare install function executed by Vue.use()
+export function install(Vue) {
+  if (install.installed) return
+  install.installed = true
+  Vue.component('MyComponent', component)
+}
+
+// Create module definition for Vue.use()
+const plugin = {
+  install
+}
+
+// Auto-install when vue is found (eg. in browser via <script> tag)
+let GlobalVue = null
+if (typeof window !== 'undefined') {
+  GlobalVue = window.Vue
+} else if (typeof global !== 'undefined') {
+  GlobalVue = global.Vue
+}
+if (GlobalVue) {
+  GlobalVue.use(plugin)
+}
+
+// To allow use as module (npm/webpack/etc.) export component
+export default component
+```
+
+Notice the first line directly imports your SFC, and the last line exports it unchanged. As indicated by the comments in the rest of the code, the wrapper provides an `install` function for Vue, then attempts to detect Vue and automatically install the component. With 90% of the work done, it's time to sprint to the finish!
+
+### How do I configure the Rollup build?
+
+With the package.json `scripts` section ready and the SFC wrapper in place, all that is left is to ensure Rollup is properly configured. Fortunately, this can be done with a small 16 line rollup.config.js file:
+
+```js
+import commonjs from '@rollup/plugin-commonjs' // Convert CommonJS modules to ES6
+import vue from 'rollup-plugin-vue' // Handle .vue SFC files
+import buble from '@rollup/plugin-buble' // Transpile/polyfill with reasonable browser support
+export default {
+  input: 'src/wrapper.js', // Path relative to package.json
+  output: {
+    name: 'MyComponent',
+    exports: 'named'
+  },
+  plugins: [
+    commonjs(),
+    vue({
+      css: true, // Dynamically inject css as a <style> tag
+      compileTemplate: true // Explicitly convert template to render function
+    }),
+    buble() // Transpile to ES5
+  ]
+}
```

Comment:
What about mentioning the `external` configuration param here?  At a minimum, almost all users are going to want `external: ['vue']`, right?  The ESM guide on freecodecamp above also doesn't mention this.

Because this guide covers all types of builds, I'm not sure how to best mention this, but I can see it being a stumbling block for readers of this guide.